addons/mrp/models/mrp_production.py

```diff
@@ -1797,6 +1808,10 @@ def button_mark_done(self):
                 'state': 'done',
             })
 
+        for production in self:
+            if production.mrp_production_backorder_count > 1 and production.picking_type_id.create_backorder == "always":
+                backorders = self.procurement_group_id.mrp_production_ids.filtered(lambda p: p.state != "done" and p.id not in self.ids)
+
```

Comment:
Why do you need that? Also I'm not sure it's a good idea to redefine backorders at each loop iteration. Probably a bug if needed